ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/EeaSendRawTransaction.java

```diff
@@ -70,6 +75,10 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
       final PrivateTransaction privateTransaction =
           PrivateTransaction.readFrom(RLP.input(Bytes.fromHexString(rawPrivateTransaction)));
 
+      final Optional<String> addPayloadEnclaveKey =
```

Comment:
I would like to refactor the respond() in a way that the code can be better understood. I find it very confusing and I think that we might miss some edge cases ... I think what I do not like is that we have to check what kind of private transaction we are handling in a lot of different places, not only in this method. 
Would be nice if the DefaultPrivacyController sendRequest() method did not to handle the privateFors/privacyGroupId stuff, but only do what it is was meant to do (check if access should be granted when multi-tenancy is enabled). 
I think it would be easier to understand and to make sure that we do not miss anything:
...
if (privateTransaction.getPrivateFor.isPresent) {
    if (isGroupAdditionTransaction) {
         // set the right precompile address, etc.
    } else {
        // normal eea private transaction, precompile contract address can be set, list of private for addresses created, etc. 
    }
}else if (privateTransaction.getPrivacyGroupId().isPresent {
    if (isOnChainPrivateTransaction) {
          // get the privateFors from on chain management contract, set the precompile contract address
    } else {
        // Orion style privacy group id, set precompile contract address
    }